crates/rmf_site_editor/src/mapf_rse/negotiation/visual.rs

```diff
@@ -0,0 +1,187 @@
+            }
+
+            let lane_material = materials.add(StandardMaterial {
+                base_color: Color::srgb(1.0, 0.0, 0.0),
+                unlit: true,
+                ..Default::default()
+            });
+
+            // Draws robot start and goal position
+            {
+                let robot_start_pos = match proposal.1.meta.trajectory.first() {
+                    Some(waypoint) => waypoint.position.translation,
+                    None => continue,
+                };
+                let robot_goal_pos = match proposal.1.meta.trajectory.last() {
+                    Some(waypoint) => waypoint.position.translation,
+                    None => continue,
+                };
+
+                let robot_start_pos =
+                    Vec3::new(robot_start_pos.x as f32, robot_start_pos.y as f32, 0.1);
+                let robot_goal_pos =
+                    Vec3::new(robot_goal_pos.x as f32, robot_goal_pos.y as f32, 0.1);
+
+                let mut spawn_circle_mesh = |pos| {
+                    commands
+                        .spawn((
+                            Mesh3d(meshes.add(Circle::new(collision_radius))),
+                            MeshMaterial3d(lane_material.clone()),
+                            Transform::from_translation(pos),
+                            Visibility::default(),
+                        ))
+                        .insert(PathVisualMarker)
+                        .insert(ChildOf(level_entity));
+                };
+                spawn_circle_mesh(robot_start_pos);
+                spawn_circle_mesh(robot_goal_pos);
+            }
+
+            let mut spawn_path_mesh = |start_pos,
+                                       end_pos,
+                                       lane_material: Handle<StandardMaterial>,
+                                       lane_mesh,
+                                       circle_mesh| {
+                commands
+                    .spawn((
+                        Mesh3d(circle_mesh),
+                        MeshMaterial3d(lane_material.clone()),
+                        Transform::from_translation(start_pos),
+                        Visibility::default(),
+                    ))
+                    .insert(PathVisualMarker)
+                    .insert(ChildOf(level_entity));
+                commands
+                    .spawn((
+                        Mesh3d(lane_mesh),
+                        MeshMaterial3d(lane_material.clone()),
+                        line_transform(&start_pos, &end_pos),
+                        Visibility::default(),
+                    ))
+                    .insert(PathVisualMarker)
+                    .insert(ChildOf(level_entity));
+            };
+
+            for (i, _waypoint) in proposal.1.meta.trajectory.iter().enumerate().skip(1) {
+                let start_pos = proposal.1.meta.trajectory[i - 1].position.translation;
+                let end_pos = proposal.1.meta.trajectory[i].position.translation;
```

Comment:
Do we need to enumerate here? It seems you are just using the index to access the array that you are iterating anyway so maybe we can use [windows](https://doc.rust-lang.org/std/primitive.slice.html#method.windows) instead?